packages/aws-cdk-lib/aws-stepfunctions/lib/state-machine.ts

```diff
@@ -460,11 +500,12 @@ export class StateMachine extends StateMachineBase {
       tracingConfiguration: props.tracingEnabled ? this.buildTracingConfiguration() : undefined,
       ...definitionBody.bind(this, this.role, props, graph),
       definitionSubstitutions: props.definitionSubstitutions,
+      encryptionConfiguration: props.encryptionConfiguration,
```

Comment:
```suggestion
      encryptionConfiguration: props.kmsKey ? {
        kmsKeyId: props.kmsKey.keyArn,
        kmsDataKeyReusePeriodSeconds: props.kmsDataKeyReusePeriod?.toSeconds(),
        type: 'CUSTOMER_MANAGED_KMS_KEY',
      } : undefined,
```
Following above suggestion. Can you please add validation on `kmsDataKeyReusePeriod` value? (must be between 60 and 900 seconds)